include/exadg/operators/generalized_laplace_operator.h

```diff
@@ -0,0 +1,781 @@
+                         GradientType const &    gradient_p,
+                         ValueType const &       value_m,
+                         ValueType const &       value_p,
+                         Vector const &          normal,
+                         CoefficientType const & coefficient)
+  {
+    ValueType const    jump_value  = value_m - value_p;
+    GradientType const jump_tensor = outer_product(jump_value, normal);
+
+    GradientType const average_gradient = 0.5 * (gradient_m + gradient_p);
+
+    return -coeff_mult(coefficient, (average_gradient - tau * jump_tensor)) * normal;
+  }
+
+  /**
+   * Same as above, but takes in as input the product of the coefficients, the gradient, and the
+   * the face normal vector. This version is for example useful for boundary integrals, where
+   * boundary conditions may prescribe the value of the this product.
+   */
+  inline DEAL_II_ALWAYS_INLINE //
+    ValueType
+    calculate_value_flux(ValueType const &       coeff_times_gradient_times_normal_m,
+                         ValueType const &       coeff_times_gradient_times_normal_p,
+                         ValueType const &       value_m,
+                         ValueType const &       value_p,
+                         Vector const &          normal,
+                         CoefficientType const & coefficient)
+  {
+    ValueType const    jump_value  = value_m - value_p;
+    GradientType const jump_tensor = outer_product(jump_value, normal);
+
+    ValueType const average_coeff_times_gradient_times_normal =
+      0.5 * (coeff_times_gradient_times_normal_m + coeff_times_gradient_times_normal_p);
+
+    return -(average_coeff_times_gradient_times_normal -
+             ValueType(coeff_mult(coefficient, (tau * jump_tensor)) * normal));
+  }
+
+  /**
+   * Calculates the interior penalty parameter `tau` for the discretization.
+   */
+  void
+  calculate_penalty_parameter(dealii::MatrixFree<dim, Number> const & matrix_free,
+                              unsigned int const                      dof_index)
+  {
+    IP::calculate_penalty_parameter<dim, Number>(penalty_parameters, matrix_free, dof_index);
+  }
+
+  void
+  calculate_coefficients(dealii::MatrixFree<dim, Number> const & matrix_free,
+                         unsigned int const                      quad_index)
+  {
+    if(data.coefficient_model == CoefficientModel::ConstantFunction)
+    {
+      constant_coefficient =
+        FunctionEvaluator<CoefficientType::rank, dim, Number>::value(*data.coefficient_function,
+                                                                     {},
+                                                                     {});
+    }
+    else if(data.coefficient_model == CoefficientModel::VariableFunction)
+    {
+      AnalyticalTensorCoefficientFunctionEvaluator<dim, Number, CoefficientType::rank>{
+        matrix_free, quad_index, data.coefficient_function, variable_coefficients}();
```

Comment:
Syntax: I find this syntax very complicated. Writing `analytical_coefficient_function.evaluate(variable_coefficients)` would be clearer / easier to understand (without need for documentation). Extracting that this line of code modifies `variable_coefficients` took me several minutes. Hence, this style of programming requires much more documentation in my opinion. I doubt this additional effort in documentation justifies the one line of code saved by this programming style.

Design: More importantly, this design does not solve my main criticsm from the last review, namely that `GeneralizedLaplace::Kernel` should in my opinion not depend on `AnalyticalTensorCoefficientFunctionEvaluator`. 

Example: When adding a new type to `CoefficientModel`, the kernel/operator has to be recompiled. I consider this an unnecessary dependency.

If we do not want to provide access to `Kernel::variable_coefficients` via a public getter returning a reference to `VariableCoefficients` (in order to fill the variable coefficients from "outside"), a lambda function with argument `VariableCoefficients &` could be passed to the kernel during construction/initialization, which allows to compute the coefficient field.

What to do if one wants to realize an analytical coefficient function that is time dependent? I believe one has to touch again `Kernel` with the current design, but this is unnecessary. Would you adjust the constructor of `AnalyticalTensorCoefficientFunctionEvaluator` and make `double time` a member variable of that class? This is probably not a good design / unnecessary, because one rather wants to call the `evaluate()` function suggested above with a second argument `time`.